pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -617,14 +617,17 @@ public void deleteFailed(ManagedLedgerException exception, Object ctx) {
             return disconnectFuture;
         }
 
-        if (producer != null && (STATE_UPDATER.compareAndSet(this, State.Starting, State.Stopping)
-                || STATE_UPDATER.compareAndSet(this, State.Started, State.Stopping))) {
+        if (producer == null) {
+            // If there's already a reconnection happening, signal to close it whenever it's ready
+            STATE_UPDATER.compareAndSet(this, State.Stopped, State.Stopped);
+        } else if (STATE_UPDATER.compareAndSet(this, State.Starting, State.Stopping)
+                || STATE_UPDATER.compareAndSet(this, State.Started, State.Stopping)) {
             log.info("[{}][{} -> {}] Disconnect replicator at position {} with backlog {}", topicName, localCluster,
                     remoteCluster, cursor.getMarkDeletedPosition(), cursor.getNumberOfEntriesInBacklog());
             return closeProducerAsync();
-        } else {
-            // If there's already a reconnection happening, signal to close it whenever it's ready
-            STATE_UPDATER.set(this, State.Stopping);
+        } else if (STATE_UPDATER.get(this) == State.Stopping) {
```

Comment:
do we really need this else block because we actually don't do anything. 
May be we can check in the beginning that if state is `Stopping` then return `CompletableFuture.completedFuture(null)`